compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -797,9 +797,9 @@ where
                 // ADTs.
                 ty::Adt(def, substs) => {
                     match this.variants {
-                        Variants::Single { index } => {
-                            TyMaybeWithLayout::Ty(def.variant(index).fields[i].ty(tcx, substs))
-                        }
+                        Variants::Single { index } => TyMaybeWithLayout::Ty(
+                            def.variant(index).fields[FieldIdx::new(i)].ty(tcx, substs),
```

Comment:
Why does this use `::new` instead of `::from_usize`?
```suggestion
                            def.variant(index).fields[FieldIdx::from_usize(i)].ty(tcx, substs),
```

Reply:
Because that file already had `VariantIdx::new`, apparently.  I'll change it since I do like the concrete one better.

(It's kinda annoying that `new` (from `Idx`), `from_usize` and `.into()` (from `newtype_index!`) all exist and work so there's nothing pushing code to consistency.)